### PR TITLE
Bump to tokio 1.0 and hyper 0.14

### PR DESCRIPTION
**Pull request by maximebedard**
Depends on https://github.com/ggriffiniii/httptest/pull/6 and https://github.com/hyperium/hyper-tls/pull/78. Will wait for version 0.14 of hyper to be pushed to crates.io and update this PR afterward with the actual release.

### CODE REVIEW

**Comment by dermesser** [comment]
Thank you. In fact, I have tried this before - https://crates.io/crates/yup-oauth2/5.0.0 - and yanked it, as there was the incompatibility with hyper you mentioned (and possibly more, I don't remember).

File: Cargo.toml [hunk 1]
```diff
@@ -14,20 +14,20 @@ edition = "2018"
 base64 = "0.12"
 chrono = { version = "0.4", features = ["serde"] }
 http = "0.2"
-hyper = "0.13.5"
-hyper-rustls = "0.20"
+hyper = { version = "0.14", features = ["client", "server", "tcp", "http2"] }
+hyper-rustls = { git = "https://github.com/ctz/hyper-rustls" }
```
Comment (dermesser) [inline Cargo.toml]: Is there a reason you use the git source here? I checked, and the version is the same as published, or are there minor fixes in git?

**Review by dermesser — CHANGES_REQUESTED** [review]
Looks good, except for one minor question

File: Cargo.toml (hunk 1 above)
Comment (hannesdejager) [reply Cargo.toml]: The master branch of hyper-rustls contains an upgrade to tokio v1.0 and hyper v.014 but this is not released yet it seems. This should probably be pinned to the revision hash if you're OK with a git ref.

Comment (dermesser) [reply Cargo.toml]: I see. Pinning would be better in any case, without me having done any research, is there a plan when the upgrade to tokio-1.0 will be released? I'd be more comfortable with staying within crates.io for now :-)

Comment (hannesdejager) [reply Cargo.toml]: Its our lucky day, after probing the author, he was kind enough to release hyper-rustls for us, you can do

`hyper-rustls = "0.22.0"`

image

Comment (maximebedard) [reply Cargo.toml]: Yes I was waiting for the released version before pinging back 😄 I've just updated it, should be good now.

**Review by dermesser — APPROVED** [review]

**Comment by dermesser** [comment]
Very nice. Given the likely breaking change of using tokio 1.0, I will release a new major version.

File: Cargo.toml (hunk 1 above)
Comment (hannesdejager) [reply Cargo.toml]: Indeed a breaking upgrade. 

hyper-rustls version was just bumped to 0.22.1 b.t.w

**Pull request merged**